File: Geo-computing/java/lab0/Stopwatch.java
package lab0;

/**
 * A timer that works like a common stopwatch.
 * @author Dave Hale, Colorado School of Mines
 * @version 2004.11.02
 */
public class Stopwatch {

  public void start() {
    if (!_running) {
      _running = true;
      _start = System.nanoTime();
    }
  }

  public void stop() {
    if (_running) {
      _time += System.nanoTime()-_start;
      _running = false;
    }
  }

  public void reset() {
    stop();
    _time = 0;
  }

  public void restart() {
    reset();
    start();
  }

  public double time() {
    if (_running) {
      return 1.0e-9*(_time+(System.nanoTime()-_start));
    } else {
      return 1.0e-9*_time;
    }
  }

  private boolean _running;
  private long _start;
  private long _time;
}

File: Geo-computing/python/lab0/stopwatch.py
#!/usr/bin/env python
import time as t


running = False 
time_ = 0.0
start_ = 0.0

def start():
    global running
    global time_ 
    global start_

    if not running:
        running = True
        start_ = t.time()

def stop(): 
    global running
    global time_ 
    global start_

    if (running) :
        time_ += t.time() - start_ 
        running = False


def reset(): 
    global running
    global time_ 
    global start_

    stop()
    time_ = 0.0

def restart():
    global running
    global time_ 
    global start_

    reset()
    start()

def time():
    global running
    global time_ 
    global start_

    if(running):
        return time_ + t.time() -start_
    else:
        return time_
